tests/perf/test_engineer_traction.py

```diff
@@ -0,0 +1,283 @@
+        bug_resp = betamax_recorder.session.get(
+            'https://bugzilla.mozilla.org/rest/bug',
+            headers={'Accept': 'application/json'},
+            params=params,
+            timeout=60,
+        )
+        return bug_resp.json()['bugs']
+
+
+@pytest.fixture
+def cooled_down_bugs(session, quantified_bugs) -> List[dict]:
+    bugs = []
+    for bug in quantified_bugs:
+        created_at = datetime.strptime(bug['creation_time'], BZ_DATETIME_FORMAT)
+        if created_at <= datetime.now() - timedelta(weeks=2):
+            bugs.append(bug)
+    return bugs
+
+
+def test_formula_demands_at_least_framework_and_suite(betamax_recorder):
+    engineer_traction = EngineerTractionFormula(betamax_recorder.session)
+
+    with pytest.raises(TypeError):
+        engineer_traction('some_framework')
+
+    with pytest.raises(TypeError):
+        engineer_traction()
+
+    with betamax_recorder.use_cassette('awsy-JS', serialize_with='prettyjson'):
+        try:
+            engineer_traction('awsy', 'JS')
+        except TypeError:
+            pytest.fail()
+
+
+def test_formula_exposes_oldest_timestamp(session):
+    engineer_traction = EngineerTractionFormula(session)
+    no_older_than = datetime.now() - timedelta(weeks=24, seconds=5)
+
+    assert engineer_traction.oldest_timestamp >= no_older_than
+
+
+@pytest.mark.parametrize(
+    'cooled_down_bug',
+    [
+        {"creation_time": "2020-05-18T15:20:55Z"},  # older than 2 weeks
+        {"creation_time": "2020-05-04T15:20:55Z"},  # older than 1 month
+        {"creation_time": "2019-12-16T08:10:37Z"},  # older than 6 months
+    ],
+)
+def test_formula_correctly_detects_cooled_down_bugs(cooled_down_bug, session):
+    engineer_traction = EngineerTractionFormula(session)
+
+    assert engineer_traction.has_cooled_down(cooled_down_bug)
+
+
+@pytest.mark.parametrize(
+    'not_cooled_down_bug',
+    [
+        {'creation_time': '2020-05-31T00:00:00Z'},  # 2 days old
+        {'creation_time': '2020-05-26T00:00:00Z'},  # 1 week old
+        {'creation_time': '2020-05-19T23:00:00Z'},  # ~2 weeks old, except for 1 hour
+    ],
+)
+def test_formula_detects_bugs_that_didnt_cool_down_yet(not_cooled_down_bug, session):
+    engineer_traction = EngineerTractionFormula(session)
+
+    assert not engineer_traction.has_cooled_down(not_cooled_down_bug)
+
+
+@pytest.mark.parametrize('bad_structured_bug', [{}, {'creation_time': 'jiberish-date'}])
+def test_formula_throws_adequate_error_for_bug(bad_structured_bug, session):
+    engineer_traction = EngineerTractionFormula(session)
+
+    with pytest.raises(ValueError):
+        engineer_traction.has_cooled_down(bad_structured_bug)
+
+
+def test_accessing_breakdown_without_prior_calculus_errors_out(session):
+    engineer_traction = EngineerTractionFormula(session)
+
+    with pytest.raises(RuntimeError):
+        _ = engineer_traction.breakdown()
+
+
+# Leveraging HTTP VCR
```

Comment:
Test code making use of betamax & betamax-serializers *(from here down to bottom of module)*